### PR TITLE
[extractor/twitter:spaces] Add scheduled time (release_timestamp) for live spaces

### PR DESCRIPTION
**Pull request by CeruleanSky**
### Description of your *pull request* and other information

Add scheduled start time for spaces

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🕒🔊🐦

Add support for extracting release timestamps from Twitter Spaces audio streams. Use the `release_timestamp` field in `yt_dlp/extractor/twitter.py` to store the scheduled start time of the stream.

> _Oh we're the coders of the Twitter Spaces_
> _We stream the audio from many places_
> _We add `release_timestamp` to the streams_
> _To schedule them ahead, so we can dream_

### Walkthrough
* Extract the scheduled start time of Twitter Spaces audio streams and convert it to Unix timestamp ([link](https://github.com/yt-dlp/yt-dlp/pull/7186/files?diff=unified&w=0#diff-8b22c3c3c6185164ce8c3e82541f91c728397dec66456eef7e05f664d5729103R1301))
* Add the extracted start time as `release_timestamp` field to the info dictionary of the audio streams ([link](https://github.com/yt-dlp/yt-dlp/pull/7186/files?diff=unified&w=0#diff-8b22c3c3c6185164ce8c3e82541f91c728397dec66456eef7e05f664d5729103R1331))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
Add/update test

File: yt_dlp/extractor/twitter.py [hunk 1]
```diff
@@ -705,6 +705,7 @@ class TwitterIE(TwitterBaseIE):
             'uploader': r're:Monique Camarra.+?',
             'uploader_id': 'MoniqueCamarra',
             'live_status': 'was_live',
+            "release_timestamp": 1658417414,
```
Comment (pukkandan) [inline yt_dlp/extractor/twitter.py:708]:
```suggestion
            'release_timestamp': 1658417414,
```

File: yt_dlp/extractor/twitter.py [hunk 2]
```diff
@@ -1298,6 +1299,7 @@ def _real_extract(self, url):
 
         metadata = space_data['metadata']
         live_status = try_call(lambda: self.SPACE_STATUS[metadata['state'].lower()])
+        scheduled_timestamp = int_or_none(metadata['scheduled_start'], scale=1000)
```
Comment (pukkandan) [inline yt_dlp/extractor/twitter.py:1302]:
```suggestion
```

File: yt_dlp/extractor/twitter.py [hunk 3]
```diff
@@ -1327,6 +1329,7 @@ def _real_extract(self, url):
             'uploader_id': traverse_obj(
                 metadata, ('creator_results', 'result', 'legacy', 'screen_name')),
             'live_status': live_status,
+            'release_timestamp': scheduled_timestamp,
```
Comment (pukkandan) [inline yt_dlp/extractor/twitter.py:1332]:
```suggestion
            'release_timestamp': try_call(
                lambda: int_or_none(metadata['scheduled_start'], scale=1000)),
```

**Review by pukkandan — APPROVED** [review]

**Comment by pukkandan** [comment]
> ###
> * [x]  At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md?rgh-link-date=2023-06-01T01%3A31%3A17Z#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)

Next time, pls actually follow this

**Comment by CeruleanSky** [comment]
Sorry about that, I did the change quick in github on the browser.   Will try to be more through next pull.

**Pull request merged**